components/api/filter/validator-filter.bal

```diff
@@ -55,27 +57,41 @@ public type validateRequestFilter object {
                     return true;
                 }
             }
-            validator:Conf|error conf = untaint validator:getAuthConfig();
-            if conf is validator:Conf {
-                string|error username = validator:validateAndGetUsername(untaint token, headerUsername ,conf);
-                if (username is string) {
-                    if username != "" {
-                        request.setHeader(constants:AUTHENTICATED_USER, username);
-                        log:printInfo(io:sprintf("The token is successfully validated for the user %s", username));
+            idp:TokenDetail|error tokenDetail;
+            if (cache.hasKey(token)) {
+                idp:TokenDetail cachedTokenDetail = <idp:TokenDetail>cache.get(token);
+                if (cachedTokenDetail.username != "") {
+                    if isExpired(cachedTokenDetail.expiryTime) {
+                        log:printError("Token is expired. Passing to the next filter");
+                        cache.remove(token);
                         return true;
                     } else {
-                        log:printError("The token is not valid");
+                        request.setHeader(constants:AUTHENTICATED_USER, cachedTokenDetail.username);
+                        log:printDebug(io:sprintf("Resolved user as %s from the cache", cachedTokenDetail.username));
                         return true;
                     }
                 } else {
-                    log:printError("When validating the token something went wrong", err = username);
+                    log:printDebug("Could not resolve username from the cache");
```

Comment:
If execution comes here doesn't that mean that the cached token details are invalid ?